src/Settings/components/vishraam.jsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from "react";
+import { Text } from "react-native";
+import { ListItem, BottomSheet, Icon, Switch } from "@rneui/themed";
+import { useSelector } from "react-redux";
+import PropTypes from "prop-types";
+import STRINGS from "../../common/localization";
+import {
+  setVishraamOption,
+  toggleVishraam,
+  setVishraamSource,
+  VISHRAAM_OPTIONS,
+  VISHRAAM_SOURCES,
+} from "../../common/actions";
+import colors from "../../common/colors";
+import styles from "../styles";
+
+const renderItem = (
+  item,
+  action,
+  name,
+  toggleVishraamOptionVisible,
+  toggleVishraamSourceVisible,
+  dispatch,
+  vishraamSource,
+  vishraamOption
+) => {
+  return (
+    <ListItem
+      key={item.key}
+      bottomDivider
+      onPress={() => {
+        toggleVishraamOptionVisible(false);
+        toggleVishraamSourceVisible(false);
+        dispatch(action(item.key));
+      }}
+    >
+      <ListItem.Content>
+        <ListItem.Title>{item.title}</ListItem.Title>
+      </ListItem.Content>
+      {name.toLowerCase() === "source" && vishraamSource === item.key && <Icon name="check" />}
+      {name.toLowerCase() === "option" && vishraamOption === item.key && <Icon name="check" />}
```

Comment:
Why are we using the key to compare with vishraamOption. Either this is a mistake or if it is done intentionally this should be changed to something that makes more sense from name.

Reply:
fixed